addons/hr_homeworking/models/hr_employee.py

```diff
@@ -180,5 +180,5 @@ def write(self, values):
             )
             for employee, locations in read_group:
                 if employee.id in employee_locations_to_remove:
-                    locations.filtered(lambda l: l.weekday in employee_locations_to_remove[employee.id]).unlink()
+                    locations.filtered(lambda l: l.weekday in employee_locations_to_remove[employee.id]).delete_default_worklocation()
```

Comment:
should not we differentiate here whether we are deleting weekly locations or normal ones? :thinking: 

Maybe we should have something like
```suggestion
normal_locations_to_unlink = self.env['hr.employee.location']
weekly_locations_to_unlink = self.env['hr.employee.location']
for employee, locations in read_group:
    locations = locations.filtered(lambda l: l.weekday in employee_locations_to_remove[employee.id]).
    normal_locations = locations.liltered(lambda l: l.weekly)
    normal_locations_to_unlink |= normal_locations
    weekly_locations_to_unlink |= locations - normal_locations
normal_locations_to_unlink.unlink()
weekly_locations_to_unlink.delete_default_worklocation()
```